### PR TITLE
[MINDEXER-176] Clean up silos

### PR DESCRIPTION
**Pull request by cstamas**
The MT updater did not clean up silos after done.

---

https://issues.apache.org/jira/browse/MINDEXER-176

### CODE REVIEW

**Comment by cstamas** [comment]
@mbien pls take a peek

**Review by mbien — COMMENTED** [review]
Makes sense to me. Thanks!

File: indexer-core/src/main/java/org/apache/maven/index/updater/IndexDataReader.java [hunk 1]
```diff
@@ -245,10 +249,18 @@ private IndexDataReadResult readIndexMT( IndexWriter w, IndexingContext context
         }
 
         LOGGER.debug( "Merging silos..." );
```
Comment (cstamas) [inline indexer-core/src/main/java/org/apache/maven/index/updater/IndexDataReader.java:251]: Adjust log msg (this is not merging, just commit+close)

File: indexer-core/src/main/java/org/apache/maven/index/updater/IndexDataReader.java [hunk 2]
```diff
@@ -245,10 +249,18 @@ private IndexDataReadResult readIndexMT( IndexWriter w, IndexingContext context
...
-        for ( IndexWriter silo : silos )
+        for ( IndexWriter siloWriter : siloWriters )
         {
-            IndexUtils.close( silo );
-            w.addIndexes( silo.getDirectory() );
+            siloWriter.commit();
+            siloWriter.close();
+        }
+        LOGGER.debug( "Cleanup of silos..." );
```
Comment (cstamas) [inline indexer-core/src/main/java/org/apache/maven/index/updater/IndexDataReader.java:257]: Adjust log msg: this is not cleanup but merge + cleanup

**Comment by mbien** [comment]
@cstamas any chance for an update release till May? I would be particularly interested in this fix here which would make me more comfortable to enable MT index extraction in NetBeans. (if not - no problem, there is always the next release :))

btw love the new formatting!

**Comment by cstamas** [comment]
Sure, am a bit distracted with resolver 1.9.5 and maven 3.9.1 but I planned to release this as next (this is NOT FINAL but a DRAFT release notes, just to make sure anon users can access it) https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12317523&version=12352530

MINDEXER-183 is still "in progress" https://github.com/apache/maven-indexer/pull/292 , will take a peek at it and merge and then release can happen.

Any hint, opinion, idea (or maybe opposing to any change above), please ping me on ML and let's discuss https://maven.apache.org/mailing-lists.html

(or hop onto ASF Slack)

**Comment by cstamas** [comment]
Oh, "upgrade to resolver 1.9.4" is also def to happen (1.9.2 has issues)

**Comment by jira-importer** [comment]
Resolve #603

**Comment by jira-importer** [comment]
Resolve #603

**Pull request merged**